packages/vite/src/node/server/index.ts

```diff
@@ -1148,7 +1148,7 @@ async function restartServer(server: ViteDevServer) {
   if (!middlewareMode) {
     await server.listen(port, true)
   } else {
-    server.ws.listen()
```

Comment:
`_createServer` is called with `hotListen: false` so `hot.listen()` is not called.
https://github.com/vitejs/vite/blob/3688f0aa225d9bfc3b2c409e19b721f7456938ba/packages/vite/src/node/server/index.ts#L944-L947